### PR TITLE
feat(handler): add error validation logic and toast for de minimis aid form (HL-1202)

### PR DESCRIPTION
**Pull request by sirtawast**
## Description :sparkles:

Add validation for handler's de minimis form when creating a new application.

### CODE REVIEW

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2933.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2933.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2933.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2933.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2933.dev.hel.ninja :laughing::tada::tada::tada:

File: frontend/benefit/handler/public/locales/en/common.json [hunk 1]
```diff
@@ -336,6 +336,10 @@
           "label": "Enimmäismäärä ylitetty",
           "content": "De minimis-tuen enimmäismäärä on ylitetty. Tuki voi olla enintään {{amount}} euroa, joka myönnetään yritykselle kuluvan vuoden ja kahden edellisen verovuoden kuluessa. Enimmäismäärässä huomioidaan kaikkien eri viranomaisten kyseisenä ajanjaksona de minimis -tukena myöntämä rahoitus."
         },
+        "deMinimisUnfinished": {
+          "label": "Puuttuvia de minimis -tuen tietoja",
+          "content": "Täytä puuttuvat minimis -kentät ja paina 'Lisää' painiketta."
```
Comment (EmiliaMakelaVincit) [inline frontend/benefit/handler/public/locales/en/common.json:341]: -  'Lisää' painiketta -> Lisää-painiketta / 'Lisää'-painiketta
- minimis -kentät -> de minimis -kentät

File: frontend/benefit/handler/src/components/applicationForm/formContent/companySection/deMinimisAid/DeMinimisAidForm.tsx [hunk 2]
```diff
@@ -36,6 +37,36 @@ const DeMinimisAidForm: React.FC<DeMinimisAidFormProps> = ({ data }) => {
...
+      formik.values.grantedAt?.length > 0,
+    [formik.values]
+  );
+
+  useEffect(() => {
+    if (isFormInputIncomplete()) {
+      setUnfinishedDeMinimisAidRow(true);
+    } else {
+      setUnfinishedDeMinimisAidRow(false);
+    }
```
Comment (EmiliaMakelaVincit) [inline frontend/benefit/handler/src/components/applicationForm/formContent/companySection/deMinimisAid/DeMinimisAidForm.tsx:67]: This effectively just equals to `setUnfinishedDeMinimisAidRow(isFormInputIncomplete())`, the if-else is redundant.

**Review by EmiliaMakelaVincit — APPROVED** [review]
Some nitpicks, can be merged without another review after possible fixes.

**Pull request merged**